src/controllers/group/index.js

```diff
@@ -50,3 +50,28 @@ const createProductIfNotGiven = async (input, models, { transaction }) => {
 
 const savePhotos = (productId, photos, { transaction }) => productPhoto
   .bulkCreate(photos.map((url) => ({ productId, url })), { transaction })
+
+export const deleteGroup = async (obj, { id }, context) => {
+  const group = await context.models.group.findByPk(id)
+
+  if (!group || await isNotDeleteable(group, context)) return 0
+
+  const productId = group.productId
+  await context.sequelize.transaction(async (t) => {
+    await context.models.groupSubscription.destroy({ where: { groupId: id } }, { transaction: t })
+    await group.destroy({}, { transaction: t })
+    await context.models.productPhoto.destroy({ where: { productId: productId } }, { transaction: t })
+    await context.models.product.destroy({ where: { id: productId } }, { transaction: t })
+  })
+
+  return 1
+}
+
+const isNotDeleteable = async (group, context) => {
+  const isSellerCorrect = context.currentUser.id === group.sellerId
```

Comment:
This check is an Authentication check and should be done at the same level where you're applying `requireAuth`. There's an utility called `combineResolvers` that you can use to combine multiple authentication checkers with the resolver that performs the required action.

Consider this check as verifying if the currently logged in user has permission to make the required change in the server.